spec/lib/rex/proto/postgresql/client_spec.rb

```diff
@@ -20,4 +20,60 @@
   end
 
   it_behaves_like 'session compatible SQL client'
+
+  describe '#map_compile_os_to_platform' do
+    [
+      { info: 'linux', expected: Msf::Platform::Linux.realname },
+      { info: 'linux2.6', expected: Msf::Platform::Linux.realname },
+      { info: 'debian-linux-gnu', expected: Msf::Platform::Linux.realname },
+      { info: 'win', expected: Msf::Platform::Windows.realname },
+      { info: 'windows', expected: Msf::Platform::Windows.realname },
+      { info: 'darwin', expected: Msf::Platform::OSX.realname },
+      { info: 'osx', expected: Msf::Platform::OSX.realname },
+      { info: 'macos', expected: Msf::Platform::OSX.realname },
+      { info: 'solaris', expected: Msf::Platform::Solaris.realname },
+      { info: 'aix', expected: Msf::Platform::AIX.realname },
+      { info: 'hpux', expected: Msf::Platform::HPUX.realname },
+      { info: 'irix', expected: Msf::Platform::Irix.realname },
+    ].each do |test|
+      it "correctly identifies '#{test[:info]}' as '#{test[:expected]}'" do
+        expect(subject.map_compile_os_to_platform(test[:info])).to eq(test[:expected])
+      end
+    end
+  end
+
+  describe '#map_compile_arch_to_architecture' do
+    [
+      { info: 'x86_64', expected: ARCH_X86_64 },
+      { info: 'x86_x64', expected: ARCH_X86_64 },
+      { info: 'x64', expected: ARCH_X86_64 },
+      { info: '64', expected: ARCH_X86_64 },
+      { info: 'x86', expected: ARCH_X86 },
+      { info: '86', expected: ARCH_X86 },
+      { info: 'i686', expected: ARCH_X86 },
+      { info: 'arm64', expected: ARCH_AARCH64 },
+      { info: 'arm', expected: ARCH_AARCH64 },
+      { info: 'sparc', expected: ARCH_SPARC },
+      { info: 'sparc64', expected: ARCH_SPARC64 },
+      { info: 'ppc', expected: ARCH_PPC },
+      { info: 'mips', expected: ARCH_MIPS },
+    ].each do |test|
+      it "correctly identifies '#{test[:info]}' as '#{test[:expected]}'" do
+        expect(subject.map_compile_arch_to_architecture(test[:info])).to eq(test[:expected])
+      end
+    end
+  end
+
+  describe '#detect_platform_and_arch' do
+    context "when the 'select version()' query is ran" do
+      it 'matches the regex' do
+        [
+          { version: '9.4', select_version_query_output: [['PostgreSQL 9.4.26 on x86_64-pc-linux-gnu (Debian 9.4.26-1.pgdg90+1), compiled by gcc (Debian 6.3.0-18+deb9u1) 6.3.0 20170516, 64-bit']] },
+          { version: '14', select_version_query_output: [['PostgreSQL 14.11 (Debian 14.11-1.pgdg120+2) on x86_64-pc-linux-gnu, compiled by gcc (Debian 12.2.0-14) 12.2.0, 64-bit']] },
+        ].each do |test|
+          expect(test[:select_version_query_output].join).to match(/.*on (\w+-\w+-\w+-\w+)/)
```

Comment:
I'm thinking this isn't a super useful test - as you're not actually calling `detect_platform_and_arch` here 👀

i.e. if the implementation of `describe '#detect_platform_and_arch' do` changed entirely and had bugs present or raised exceptions for your expected inputs, these tests would still pass

Potentially:

```ruby
  describe '#detect_platform_and_arch' do
    [
      { version: 'PostgreSQL 9.4.26 on x86_64-pc-linux-gnu (Debian 9.4.26-1.pgdg90+1), compiled by gcc (Debian 6.3.0-18+deb9u1) 6.3.0 20170516, 64-bit', expected: 'TODO' },
      { version: 'PostgreSQL 14.11 (Debian 14.11-1.pgdg120+2) on x86_64-pc-linux-gnu, compiled by gcc (Debian 12.2.0-14) 12.2.0, 64-bit', expected: 'TODO' },
    ].each do |test|
      context "when the database is version #{test[:version]}" do
        it "returns #{test[:expected]}" do
          mock_query_result = instance_double DatabaseResultClassYouWillNeedToFindThisYourself, rows: [[test[:version]]
          allow(subject).to receive(:query).with('select version()').and_return(mock_query_result)

          expect(subject.detect_platform_and_arch).to eq test[:expected]
        end
      end
    end
  end

```

(Not tested, and you'll need to slot in some of the blanks)